### ISSUE TITLE
[Feature Request] include "VS Code" in standard windows workspace template

### ISSUE THREAD

**Issue by sxinger** [Missouri-BMI/service-workbench-on-aws#39]
As l learned from more swb users, it might be useful to include a python IDE in the standard windows workspace template. Although sagemaker provides an alternative web-based interface for coding in python, but many users are more used to the desktop setting for easy navigation to other components. Between Pycharm and VS Code, I vote VS Code for its being open source and lightweight. 

In reflection on https://github.com/Missouri-BMI/NextGen_Cloud_Architecture_AWS/issues/102, I would also like to try model the best practice for software installation request. 

----------------------------------------------------------
Requester: Xing Song
Software: VS Code IDE
Target Workspace Template: Standard Windows (https://github.com/Missouri-BMI/NextGen_Cloud_Architecture_AWS/wiki/AWS-SWB-Workspace-Types)
- recommended installer source: https://code.visualstudio.com/Download 

------------------------------------------------------------------------
- [ ] DevOps review by @vasanthi014 
- [ ] InfoSec review by @etanye 
- [ ] SysAdmin review @fergusonsha-umkc

**Comment by etanye** [comment]
Is Python currently available in the workspace template? My only possible concern is lateral network discovery using a tool such as Python. @fergusonsha-umkc please confirm: is each EC2 instance in it's own VPC, or is each in a separate security group?

@sxinger Including VS code in the workspace template is fine.

**Comment by fergusonsha-umkc** [comment]
each EC2 instance is in a separate security group, yes.  No lateral communication permissible between instances within the subnet.

**Comment by etanye** [comment]
@sxinger Yes, please check the box. I would have checked it but it is grayed out for me.

**Comment by sxinger** [comment]
@vasanthi014 can you review this request? Please feel free to suggest anyone else from your team to be the point person for such review?

**Comment by vasanthi014** [comment]
I am okay with having VS Code in the template. 

We might need to explore if the free version of pycharm can also be installed if needed. @kzraryan-mu Do you use pycharm in SWB as well?

**Comment by sxinger** [comment]
> I am okay with having VS Code in the template.
> 
> We might need to explore if the free version of pycharm can also be installed if needed. @kzraryan-mu Do you use pycharm in SWB as well?

I would pick one between pycharm and vs code, but not both.

**Comment by vasanthi014** [comment]
Should we use [SWB Software Request] instead of [Feature Request]. I like the format Song used with Requester name, Software and Target Workspace Template. I can create a github issue template once we decide on the issue structure.

**Comment by sxinger** [comment]
@kzraryan-mu  care to weigh in? I'm fine with either pycharm or vs code

**Comment by sxinger** [comment]
> Should we use [SWB Software Request] instead of [Feature Request]. I like the format Song used with Requester name, Software and Target Workspace Template. I can create a github issue template once we decide on the issue structure.

I was using the built-in template inherited from the main https://github.com/awslabs/service-workbench-on-aws branch, I like the idea of creating our own template with the title:  [SWB Software Request]. Let's do that.

**Comment by vasanthi014** [comment]
> @kzraryan-mu care to weigh in? I'm fine with either pycharm or vs code

pycharm is free for students, faculty and academic staff https://www.jetbrains.com/community/education/#students 
Side note: Rana is a big fan of everything JetBrains creates :) 

I will let @kzraryan-mu weigh in on pycharm or vs code

**Comment by sxinger** [comment]
besides I've recently become a fun of chatGPT (microsoft), I also found this post: https://blog.logrocket.com/pycharm-vs-vscode/. 

Another thought: is pycharm integratable with snowflake? https://docs.snowflake.com/en/user-guide/vscode-ext.html

**Comment by vasanthi014** [comment]
> besides I've recently become a fun of chatGPT (microsoft), I also found this post: https://blog.logrocket.com/pycharm-vs-vscode/.
> 
> Another thought: is pycharm integratable with snowflake? https://docs.snowflake.com/en/user-guide/vscode-ext.html

Yes it is, You can connect to snowflake and also github from pycharm

**Comment by fergusonsha-umkc** [comment]
I am actually not opposed to adding both VSCode and PyCharm Edu Edition... however I would be very hesitant with using chatGPT for anything in this environment. @etanye what do you think?

**Comment by sxinger** [comment]
> I am actually not opposed to adding both VSCode and PyCharm Edu Edition... however I would be very hesitant with using chatGPT for anything in this environment. @etanye what do you think?

Oh, no, I didn't mean to add chatGPT, just want to say that I have currently become a fun of Microsoft because of chatGPC and vs code is a microsoft product! Sorry about the confusion!

In terms of memory and storage consumption, I would want to minimize software redundancy.

**Comment by fergusonsha-umkc** [comment]
Yep, can make sure we won't need to grow disk for this.  Shouldn't need to, but good to verify :)
I have an idea about how to make this a user choice after they connect, to install one or both as they prefer... I'll need to test it before promising anything.  Concept being: desktop shortcuts to self-cleaning PowerShell scripts that user can launch to install their IDE of choice, on-demand.  Would also shorten time to completion of the deployment phase...

**Comment by kzraryan-mu** [comment]
@vasanthi014 @sxinger 
I have seen the ChatGPT integration with VS Code. And looking for similar in PyCharm. I like PyCharm as I was always more inclined to the diverse software of IntelliJ. VS Code and IntelliJ both have most of the automation we need. 

If we add IDE for general purposes, either one works fine (as many researchers/students might like one or another). But for our development/SWB, I would really like us to stick to one. And I would vote for PyCharm (IntelliJ). There are so many reasons, but discussing them in person is better.

**Comment by sxinger** [comment]
Let's do both then. But let's do the review for one single software per ticket. This one is only for VS Code. 

@vasanthi014 can you create another similar request ticket for PyCharm installation on standard swb windows template?

**Comment by kzraryan-mu** [comment]
@vasanthi014 If we don't use the Professional Version (free for academia), We will not get the benefits we need. As we are installing a free version of VS Code, let's install the professional version of PyCharm.

**Comment by sxinger** [comment]
@fergusonsha-umkc I think we are good with moving to the "Resolve" phase by adding "VS Code" to the standard windows workspace
